### ISSUE TITLE
some thing error in build avatar-panda

### ISSUE THREAD

**Issue by ZakiG** [avatartwo/avatar-panda#2]
Hi,
I want to build avatar-panda. But when I use `make -j4`, I got those strange error:
`
  API     arm-softmmu/panda/plugins/osi/osi_ext.h 
  API     arm-softmmu/panda/plugins/callstack_instr/callstack_instr_ext.h 
Traceback (most recent call last):
  File "/home/XXXXXX/ndss18_wycinwyc/panda_modifications/avatar-panda/panda/scripts/pp.py", line 86, in <module>
    if (p[0] == '#'): continue
IndexError: string index out of range
/home/XXXXXX/ndss18_wycinwyc/panda_modifications/avatar-panda/panda/Makefile.panda.target:68: recipe for target 'plog.proto' failed
make[1]: *** [plog.proto] Error 1
Makefile:324: recipe for target 'subdir-arm-softmmu' failed
`
My installation process is followed the `build_panda_wycinwyc.sh` and I believe all dependencies  are installed. 
However , when I build `PANDA`, the compilation process is fine. 
Best,
Zaki

**Comment by mariusmue** [comment]
I agree with the strangeness  of this bug and have no explanation by hand.
Can you describe more your build environment? Over here, avatar-panda builds just fine.

Also, as you were following  ``build_panda_wycinwyc.sh` - did you verify that you copied over and modified the Makefiles in the right way?

Sorry for all the complications.

Best,
Marius

**Comment by LittleSec** [comment]
I met this error, too. Ubuntu16.04, and run `build_panda_wycinwyc.sh` directly. (In fact, I fix line 23 `git submodule update --init avatar-panda` to `git clone https://github.com/avatartwo/avatar-panda.git --depth=1`, because I try some times and can't not git clone avatar-panda by former command).
And I fix the `pp.py` file to build continue: 
```python
if (len(p) == 0 or p[0] == '#'): 
        continue
```
By the way, in line 101 has the same python code, but I don't fix that.
Maybe run the  `build_panda_wycinwyc.sh` script will has some problems? but can according to the messages fix them.
